### PR TITLE
[embedded] Detect WMO via .hasPrimaryInputs() instead of -wmo flag presence

### PR DESCRIPTION
**Pull request by kubamracek**

### CODE REVIEW

**Comment by kubamracek** [comment]
@swift-ci please test

**Comment by kubamracek** [comment]
@swift-ci please test

File: lib/Frontend/CompilerInvocation.cpp [hunk 1]
```diff
@@ -1323,6 +1323,11 @@ static bool ParseLangArgs(LangOptions &Opts, ArgList &Args,
       Diags.diagnose(SourceLoc(), diag::evolution_with_embedded);
       HadError = true;
     }
+
+    if (FrontendOpts.InputsAndOutputs.hasPrimaryInputs()) {
```
Comment (CodaFi) [inline lib/Frontend/CompilerInvocation.cpp:1327]: I don’t really understand this. Primaries and WMO don’t really mix. Are you trying to detect the optimization mode, or the compilation mode?

Comment (kubamracek) [reply lib/Frontend/CompilerInvocation.cpp:1327]: Thanks for chiming in here, I could use some help with this :)

What I’m trying to do is to require WMO, the compilation mode. I thought that the presense of a primary file is a clear sign that we’re *not* doing WMO (and vice versa). Is that not true?

**Comment by eeckstein** [comment]
WMO checking is already done by https://github.com/apple/swift/pull/68358, right?

**Comment by kubamracek** [comment]
Yes, but I realized that the -wmo flag is a flag for the driver, and not for the frontend. So https://github.com/apple/swift/pull/68358 isn't going to work if we actually compile with the driver. This is the whole point of this PR, I should have started with stating that, sorry! :)

**Comment by kubamracek** [comment]
@swift-ci please test and merge

**Pull request merged**